File: Proyectos/ms. Pacman/MsPacman/src/app/AgentProgramm.java
package app;

import java.util.PriorityQueue;

public class AgentProgramm {

    private Board board;

    public AgentProgramm(Byte[][] board) {
        this.board = new Board(board);
    }

    public void aStar(){

        PriorityQueue<Board> queue = new PriorityQueue<Board>();
        Board aux_board;

        while(!board.finished()){
            for (int i = 0; i < Moves.moves.length; i++) {
                aux_board = Moves.moves[i].move(board.clone());
                if(aux_board == null) continue;

                aux_board.setPriority(Heuristic.heuristicA(aux_board));
                aux_board.printBoard();
                queue.offer(aux_board);
            }

            board = queue.poll();
            //board.printBoard();
            System.out.println(board.getLastMovement());
            queue.clear();
        }
        System.out.println("Completado");

    }

    public Board getBoard() {
        return this.board;
    }

    public void setBoard(Board board) {
        this.board = board;
    }
}

File: Proyectos/pacman/agent_programm.py
from board import Board

from moves import moves
from heuristic import Heuristic

from Queue import PriorityQueue
import copy

class AgentProgramm:

    def __init__(self, board):
        self.board = Board(board)

    def a_star(self):

        heuristic = Heuristic()
        moves_solution = []
        print

        while not self.board.num_not_visited == 0 and not self.board == None:
            queue = PriorityQueue()
            for move in moves:
                aux_board = move(copy.deepcopy(self.board))
                if aux_board == None:
                    continue

                aux_board.priority = heuristic.heuristicA(aux_board)
                queue.put(aux_board)

            self.board = queue.get()
            # self.board.printBoard()
            moves_solution.append(self.board.last_movement)
            # queue.queue.clear()

        print('finish with a total of {} moves').format(len(moves_solution))
    
